File: generar.js
const fs = require('fs');
const path = require('path');

function listarArchivos(dir, base = '') {
    let resultados = [];
    const items = fs.readdirSync(dir);
    for (const item of items) {
        const fullPath = path.join(dir, item);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            resultados = resultados.concat(listarArchivos(fullPath, path.join(base, item)));
        } else {
            const ext = path.extname(item).toLowerCase();
            if (['.mp3', '.wav', '.ogg'].includes(ext)) {
                resultados.push({
                    titulo: path.basename(item, ext),
                    archivo: path.join(base, item).replace(/\\/g, '/'),
                    album: 'Carpeta local'
                });
            }
        }
    }
    return resultados;
}

const canciones = listarArchivos('Musica'); // Cambia si es 'Música'
fs.writeFileSync('lista.json', JSON.stringify(canciones, null, 2));
console.log(`✅ lista.json generado con ${canciones.length} canciones.`);

File: generar.py
import os
import json

def listar_archivos(dir, base=''):
    resultados = []
    for item in os.listdir(dir):
        full = os.path.join(dir, item)
        if os.path.isdir(full):
            # Recursivo: la subcarpeta se usa como "álbum"
            resultados += listar_archivos(full, os.path.join(base, item))
        else:
            ext = os.path.splitext(item)[1].lower()
            if ext in ['.mp3', '.wav', '.ogg']:
                # El álbum es el nombre de la subcarpeta (última parte de base)
                album = os.path.basename(base) if base else 'Raíz'
                resultados.append({
                    'titulo': os.path.splitext(item)[0],
                    'archivo': os.path.join(base, item).replace('\\', '/'),
                    'album': album
                })
    return resultados

# Cambia "Musica" por el nombre exacto de tu carpeta raíz (sin acento o con él)
canciones = listar_archivos('Musica')

# Guarda el JSON
with open('lista.json', 'w', encoding='utf-8') as f:
    json.dump(canciones, f, indent=2, ensure_ascii=False)

print(f'✅ lista.json generado con {len(canciones)} canciones.')